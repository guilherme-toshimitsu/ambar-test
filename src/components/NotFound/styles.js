const styles = {
  container: {
    width: '100%',
    // height: '100%',
    minHeight: 317,
    backgroundColor: '#ffffff',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  iconBg: {
    width: 116,
    height: 116,
    backgroundColor: '#0086ff',
    backgroundColor: 'rgba(0, 134, 255, 0.2)',
    borderRadius: 50,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 'auto',
  },
  icon: {
    width: 40,
    height: 36,
    color: '#0086ff',
  },
  text: {
    width: 307,
    height: 19,
    fontSize: 14,
    color: '#404040',
  },
};
export default styles;
