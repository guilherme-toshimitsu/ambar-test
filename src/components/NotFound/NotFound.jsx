import React from 'react';

import { ReactComponent as SearchIcon } from '../../assets/icons/search.svg';
import PropTypes from 'prop-types';

import { makeStyles } from '@material-ui/core';

import styles from './styles';

const useStyles = makeStyles(styles);

const NotFound = ({ text }) => {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      <div>
        <div className={classes.iconBg}>
          <SearchIcon className={classes.icon} />
        </div>
        <p className={classes.text}>{text}</p>
      </div>
    </div>
  );
};

NotFound.defaultProps = {
  text: 'Nenhum resultado encontrado para essa busca',
};

NotFound.propTypes = {
  text: PropTypes.string,
};

export default NotFound;
