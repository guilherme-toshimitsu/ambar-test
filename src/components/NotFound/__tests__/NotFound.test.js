import React from 'react';
import { render } from '@testing-library/react';
import NotFound from '../NotFound';

describe('<NotFound> Screen for the show when not found a list seller', () => {
  it('renders', () => {
    const container = render(<NotFound text={'text'} />);
    expect(container).toMatchSnapshot();
  });
});
