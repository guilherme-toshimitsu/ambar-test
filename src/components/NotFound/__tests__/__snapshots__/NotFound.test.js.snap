// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound> Screen for the show when not found a list seller renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-container-1"
      >
        <div>
          <div
            class="makeStyles-iconBg-2"
          >
            <svg
              class="makeStyles-icon-3"
            >
              search.svg
            </svg>
          </div>
          <p
            class="makeStyles-text-4"
          >
            text
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-container-1"
    >
      <div>
        <div
          class="makeStyles-iconBg-2"
        >
          <svg
            class="makeStyles-icon-3"
          >
            search.svg
          </svg>
        </div>
        <p
          class="makeStyles-text-4"
        >
          text
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
